Argument order issue test

cryptix opened the pull request:
see #190

cznic commented on driver/weird_test.go:
```diff
@@ -0,0 +1,111 @@
+	"testing"
+)
+
+func check(err error, t *testing.T) {
```
I can reproduce the failure, but using this function hides where it actually happens. I can recover the information by eg. inserting a panic here, but please use `t.Fail*` directly at the failure place and remove this function, thank you.

cznic commented on driver/weird_test.go:
```diff
@@ -0,0 +1,111 @@
+
+	// prepare db
+	tx, err := db.Begin()
+	check(err, t)
```
I'm aware that replacing all this nice single line checks with at least 3 lines is a bit painful, but please use old fashioned

```go
if err != nil {
        t.Fatal(err)
}
```

everywhere, thank you.

cznic commented on driver/weird_test.go:
```diff
@@ -0,0 +1,111 @@
+	err = tx.Commit()
+	check(err, t)
+	if cnt != 1 {
+		fmt.Printf("affected: %d\n", cnt)
```
Please `s/fmt.Printf/t.Logf/`, assuming it's not an error actually.

cryptix commented:
I removed the `check(err,t)` as requested and will try to move it out of the `driver` package. I just added it there since it was simple to copy my reproduction in there. I haven't used ql without the driver yet but looking at the examples in `all_test.go` it looks straight forward to use, just need to convert the statements.

> Glad to learn more about what you've possibly already figured out about this.

Not much more to add right now other than it feels  _off-by-one_-ish. `Exec(float, id, string)` also leads to the error for instance. I just had luck finding a combination that worked.

cznic commented:
> I haven't used ql without the driver yet but looking at the examples in all_test.go it looks straight forward to use, just need to convert the statements.

Feel free to keep it where it's right now or move one directory up. I'd prefer the parent directory but it's not critical.

cznic commented:
BTW, you surely _can_ use the driver from the parent directory. No need to rewrite the test.

cznic commented:
Example: https://github.com/cznic/ql/blob/8c32ff10fdc2810e93a6e127119d95545497909e/all_test.go#L2005

cryptix commented:
Feel free to merge #192. These 7 commits felt to noisy just for the reproduction of this bug.
